### PR TITLE
Make weave memory resources configurable

### PR DESCRIPTION
**Pull request by itskingori**
It's pretty clear that the 200MB doesn't suffice for large clusters (even when less than 100 node). Here's an example of my 7-day experience 👇

Screenshot 2019-07-23 at 16 01 56

We manually set it to `300MB` just to make sure our weave pods don't get OOM'd. We currently have `CONN_LIMIT` to the default of 100 but its worth noting increasing this limit means there'll be an increase in the memory consumption of the weave container in the weave pod.

~Will add auto-generated code after comments and feedback.~

### CODE REVIEW

**Comment by itskingori** [comment]
@bboreham @murali-reddy Just tagging you, FYI.

File: upup/models/cloudup/resources/addons/networking.weave/k8s-1.12.yaml.template [hunk 1]
```diff
@@ -172,9 +172,17 @@ spec:
...
               cpu: 50m
+              {{- if .Networking.Weave.MemoryResources }}
+              memory: {{ .Networking.Weave.MemoryResources }}
+              {{else}}
               memory: 200Mi
+              {{- end }}
             limits:
+              {{- if .Networking.Weave.MemoryResources }}
+              memory: {{ .Networking.Weave.MemoryResources }}
+              {{else}}
```
Comment (zacblazic) [inline upup/models/cloudup/resources/addons/networking.weave/k8s-1.12.yaml.template]: Could be wrong, but you may need to add `-` here for whitespace control. Not sure what the output looks like though.

Comment (itskingori) [reply upup/models/cloudup/resources/addons/networking.weave/k8s-1.12.yaml.template]: Ah, thanks. Didn't know about that. Fixing.

Comment (itskingori) [reply upup/models/cloudup/resources/addons/networking.weave/k8s-1.12.yaml.template]: Checked other templates and just did it how they've done it there. See https://github.com/kubernetes/kops/pull/7310/commits/ff20844db2bb4db951d8eb264803d45c06d3fc66.

**Comment by bboreham** [comment]
Can I suggest you also look at the stats for paging - `rate(container_memory_failures_total{scope="container",failure_type="pgmajfault"}[1m])` - because this is potentially worse than OOM when it comes to a network daemon.

Did you consider the ["addon resizer"](https://github.com/kubernetes/autoscaler/tree/master/addon-resizer)? (something I only recently discovered)

**Pull request closed without merging**